facebook/OpenBIC
# fby4: wf: Fix abnormal CCI message when DC off

# Description
- The root cause is that we put down the CXL-ready flag at the wrong time. Thus, BIC was still reading the DIMM sensor through MCTP/CCI from CXL. But CXL is executed DC off at that time.
Solution is that move the change time of putting down CXL-ready flag.
- Separate function of access checking for CXL1 and CXL2.

# Motivation
- BIC console show MCTP/CCI command failed when host DC off.

# Test plan
- Build code: Pass
- Check BIC console: Pass

# Log
- Before fix uart:~$ [00:06:47.611,000] <inf> plat_power_seq: MB DC (gpio num 7) status is OFF [00:06:47.611,000] <wrn> power_status: DC_STATUS: off [00:06:48.582,000] <err> hal_i2c: I2C 3 master write retry reach max with ret -6 [00:06:48.582,000] <err> mctp: i2c_master_write failed, ret -6 [00:06:48.582,000] <wrn> mctp: mctp write data failed [00:06:48.582,000] <wrn> cci: mctp_send_msg error!! [00:06:48.582,000] <wrn> cci: send msg failed!
[00:06:48.582,000] <err> hal_i2c: I2C 3 master write retry reach max with ret -6 [00:06:48.583,000] <err> mctp: i2c_master_write failed, ret -6 [00:06:48.583,000] <wrn> mctp: mctp write data failed [00:06:48.583,000] <wrn> cci: mctp_send_msg error!! [00:06:48.583,000] <wrn> cci: send msg failed!
[00:06:48.584,000] <err> hal_i2c: I2C 3 master write retry reach max with ret -6 ...
[00:06:51.597,000] <err> hal_i2c: I2C 3 master write retry reach max with ret -6 uart:~$ 00:06:51.598,000] <err> pldm_sensor: Failed to read sensor_num 0x6 of thread 4, status0x4 [00:06:51.597,000] <err> mctp: i2c_master_write failed, ret -6 uart:~$  <inf> plat_power_seq: CXL 1 power off success [8D[00:06:51.597,000] <wrn> mctp: mctp write data failed [00:06:51.597,000] <wrn> cci: mctp_send_msg error!! [00:06:51.597,000] <wrn> cci: send msg failed!
[00:06:51.598,000] <err> hal_i2c: I2C 3 master write retry reach max with ret -6 [00:06:51.598,000] <err> mctp: i2c_master_write failed, ret -6 [00:06:51.598,000] <wrn> mctp: mctp write data failed [00:06:51.598,000] <wrn> cci: mctp_send_msg error!! [00:06:51.598,000] <wrn> cci: send msg failed!
[00:06:51.598,000] <wrn> cci: Retry reach max!
[00:06:51.598,000] <err> vistara: CCI command 0xc531 read fail [00:06:51.598,000] <err> pldm_sensor: Failed to read sensor_num 0x6 of thread 4, status0x4 uart:~$ [00:06:56.013,000] <inf> plat_power_seq: CXL 2 power off success

- After fix uart:~$ [00:02:00.315,000] <inf> plat_power_seq: MB DC (gpio num 7) status is OFF [00:02:13.453,000] <wrn> power_status: DC_STATUS: off uart:~$ mctp ctrl msg timeout!!
cmd 1, inst_id 0
[00:02:17.654,000] <inf> plat_power_seq: CXL 1 power off success [00:02:21.854,000] <inf> plat_power_seq: CXL 2 power off success
